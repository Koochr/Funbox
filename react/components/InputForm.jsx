import React from "react"
import PropTypes from "prop-types"

export default class InputForm extends React.Component {
	static propTypes = {
		onSubmit: PropTypes.func.isRequired,
	}

	state = {
		value: "",
	}

	handleChange = event => {
		this.setState({ value: event.target.value, })
	}

	handleSubmit = () => {
		this.props.onSubmit(this.state.value)
		this.setState({ value: "", })
	}

	onKeyPress = event => {
		if (event.charCode === 13) {
			event.preventDefault()
			this.handleSubmit()
		}
	}

	render() {
		return <React.Fragment>
			<form className="input-form">
				<input
					className="input-field form-control"
					type="text"
					onChange={this.handleChange}
					value={this.state.value}
					onKeyPress={this.onKeyPress}
				/>
				<button className="btn btn-warning" type="button" onClick={this.handleSubmit}>Add</button>
			</form>
		</React.Fragment>
	}
}
