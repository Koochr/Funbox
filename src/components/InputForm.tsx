import * as React from 'react';

interface InputFormProps {
  onSubmit: (name: string) => void;
}

const InputForm: React.FunctionComponent<InputFormProps> = props => {
  const [value, setValue] = React.useState('');

  const handleChange: (event: React.SyntheticEvent) => void = event => {
    const target = event.target as HTMLInputElement;
    setValue(target.value);
  };

  const handleSubmit: () => void = () => {
    props.onSubmit(value);
    setValue('');
  };

  const onKeyPress: (event: React.KeyboardEvent) => void = event => {
    if (event.charCode === 13) {
      event.preventDefault();
      handleSubmit();
    }
  };

  return (
    <React.Fragment>
      <form className="input-form">
        <input
          className="input-field form-control"
          type="text"
          onChange={handleChange}
          value={value}
          onKeyPress={onKeyPress}
        />
        <button
          className="btn btn-warning"
          type="button"
          onClick={handleSubmit}
        >
          Add
        </button>
      </form>
    </React.Fragment>
  );
};

export default InputForm;
