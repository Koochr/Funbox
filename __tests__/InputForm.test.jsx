/* global it, jest, expect */
import React from "react"
import { shallow, configure, mount, } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import InputForm from "../react/components/InputForm.jsx"

configure({ adapter: new Adapter(), })

it("Should render correctly", () => {
	const component = shallow(
		<InputForm onSubmit={() => {}}/>
	)
	expect(component).toMatchSnapshot()
})

it("Should render needed components", () => {
	const component = mount(
		<InputForm onSubmit={() => {}} />
	)
	const form = component.find("form")
	expect(form).toHaveLength(1)
	const input = form.find("input")
	expect(input).toHaveLength(1)
	const button = form.find("button")
	expect(button).toHaveLength(1)
})

it("Should handle input change and submit by button press", () => {
	const component = mount(
		<InputForm onSubmit={jest.fn()}/>
	)
	const enteredText = "some text"
	expect(component.state().value).toStrictEqual("")
	component.find("input").simulate("change", { target: { value: enteredText, }, })
	expect(component.state().value).toStrictEqual(enteredText)
	component.find("button").simulate("click")
	expect(component.props().onSubmit).toBeCalledWith(enteredText)
})

it("Should handle input change and submit by enter key press", () => {
	const component = mount(
		<InputForm onSubmit={jest.fn()}/>
	)
	const enteredText = "other text"
	expect(component.state().value).toStrictEqual("")
	component.find("input").simulate("change", { target: { value: enteredText, }, })
	expect(component.state().value).toStrictEqual(enteredText)
	component.find("input").simulate("keyPress", { charCode: 13, })
	expect(component.props().onSubmit).toBeCalledWith(enteredText)
})
