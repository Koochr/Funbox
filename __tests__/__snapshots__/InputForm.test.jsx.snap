// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputForm
    onSubmit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <form
        className="input-form"
      >
        <input
          className="input-field form-control"
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
        <button
          className="btn btn-warning"
          onClick={[Function]}
          type="button"
        >
          Add
        </button>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="input-field form-control"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value=""
          />,
          <button
            className="btn btn-warning"
            onClick={[Function]}
            type="button"
          >
            Add
          </button>,
        ],
        "className": "input-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-field form-control",
            "onChange": [Function],
            "onKeyPress": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add",
            "className": "btn btn-warning",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Add",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <form
          className="input-form"
        >
          <input
            className="input-field form-control"
            onChange={[Function]}
            onKeyPress={[Function]}
            type="text"
            value=""
          />
          <button
            className="btn btn-warning"
            onClick={[Function]}
            type="button"
          >
            Add
          </button>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="input-field form-control"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="text"
              value=""
            />,
            <button
              className="btn btn-warning"
              onClick={[Function]}
              type="button"
            >
              Add
            </button>,
          ],
          "className": "input-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input-field form-control",
              "onChange": [Function],
              "onKeyPress": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add",
              "className": "btn btn-warning",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Add",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
